File: src/nodes/push_icp_ds_action.cpp
#include <ros/ros.h>
#include <pal_locomotion_msgs/ActionWithParameters.h>
#include <pal_locomotion_msgs/PushActions.h>
#include <property_bag/property_bag.h>
#include <pal_locomotion/step.h>
#include <boost/archive/text_oarchive.hpp>
#include <boost/archive/text_iarchive.hpp>
#include <property_bag/serialization/property_bag_boost_serialization.h>

using namespace pal_locomotion;

int main(int argc, char **argv)
{
  // Set up ROS.
  ros::init(argc, argv, "move_icp_action_start");
  ros::NodeHandle nh;

  ros::ServiceClient client = nh.serviceClient<pal_locomotion_msgs::PushActions>(
      "/biped_walking_dcm_controller/push_actions");

  pal_locomotion_msgs::PushActions push_actions_request;

  pal_locomotion_msgs::ActionWithParameters new_action;
  new_action.action_type = "pal_locomotion::MoveICPDSAction";
  property_bag::PropertyBag parameters;

  std::stringstream ss;
  boost::archive::text_oarchive oa(ss);
  oa << parameters;
  new_action.action_parameters = ss.str();
  push_actions_request.request.actions.push_back(new_action);

  if (client.call(push_actions_request))
  {
    ROS_INFO("Succesfully called service");
  }
  else
  {
    ROS_ERROR("Failed to call service");
    return 1;
  }
}

File: src/scripts/push_icp_ds_action.py
import sys
import rospy


from pal_locomotion_msgs.srv import PushActions, PushActionsRequest, PushActionsResponse
from pal_locomotion_msgs.msg import ActionWithParameters


rospy.init_node('push_icp_ds_action_node', anonymous=True)

push_action_service = rospy.ServiceProxy('/biped_walking_dcm_controller/push_actions', PushActions)

new_action = ActionWithParameters(action_type='pal_locomotion::MoveICPDSAction', action_parameters='')

request = PushActionsRequest(actions=[new_action])
print ('requst',request)
response = push_action_service(request)
print ('response', response)
